tests/test_user_security.py

```diff
@@ -0,0 +1,200 @@
+import enum
+import unittest
+from datetime import timedelta, datetime
+from http import HTTPStatus
+
+import jwt
+from django.conf import settings
+from django.http import HttpRequest
+from sqlalchemy import Integer, Column, VARCHAR
+
+from core_lib.data_layers.data.db.sqlalchemy.base import Base
+from core_lib.data_layers.data_access.db.crud.crud import CRUD
+from core_lib.data_layers.data_access.db.crud.crud_data_access import CRUDDataAccess
+from core_lib.data_transform.result_to_dict import result_to_dict
+from core_lib.rule_validator.rule_validator import ValueRuleValidator, RuleValidator
+from core_lib.session.jwt_token_handler import JWTTokenHandler
+from core_lib.session.security_handler import SecurityHandler
+from core_lib.session.user_security import UserSecurity
+from core_lib.web_helpers.decorators import handle_exceptions
+from core_lib.web_helpers.django.decorators import RequireLogin
+from core_lib.web_helpers.request_response_helpers import response_status
+from core_lib.web_helpers.web_helprs_utils import WebHelpersUtils
+from tests.test_data.test_utils import connect_to_mem_db
+
+
+# CRUD SETUP
+class User(Base):
+    __tablename__ = 'user_security'
+
+    id = Column(Integer, primary_key=True, nullable=False)
+    username = Column(VARCHAR(length=255), nullable=False, default="")
+    email = Column(VARCHAR(length=255), nullable=False, default="")
+    role = Column(Integer, nullable=False)
+
+
+class UserDataAccess(CRUDDataAccess):
+    allowed_update_types = [
+        ValueRuleValidator('username', str),
+        ValueRuleValidator('email', str),
+        ValueRuleValidator('role', int),
+    ]
+
+    rules_validator = RuleValidator(allowed_update_types)
+
+    def __init__(self):
+        CRUD.__init__(self, User, connect_to_mem_db(), UserDataAccess.rules_validator)
+
+
+# USER SECURITY SETUP
+class PolicyRoles(enum.Enum):

```

Comment:
move to line 29

Reply:
I have kept it outside the class for the reason I need to access it in other parts of the test